### ISSUE TITLE
Abstract methods can only appear within an abstract class

### ISSUE THREAD

**Issue by mabab** [ember-intl/ember-intl#1521]
- [ yes] I am on the [latest](https://github.com/ember-intl/ember-intl/releases/latest) ember-intl version
- [ yes] I have searched the [issues](https://github.com/ember-intl/ember-intl/issues) of this repo and believe that this is not a duplicate

### Environment

- Ember Version: 3.24.0
- Ember CLI Version: 3.24.0
- Ember Intl Version: 5.6.2
- Browser(s):  Google Chrome 87.0.4280.88
- Node Version: 14.9.0

### Steps to Reproduce

1.  Create new ember project -> ember new pet-project
2. Install ember-intel -> ember install ember-intl
3. start server -> ember server
 
 Build Error (broccoli-persistent-filter:Babel > [Babel: ember-intl]) in ember-intl/-private/formatters/-base.ts

/home/mabab/my-project/signer/test-intel/ember-intl/-private/formatters/-base.ts: Abstract methods can only appear within an abstract class. (36:2)

  34 |   static type: keyof Formats;
  35 |
> 36 |   protected abstract readonly createNativeFormatter: (
     |   ^
  37 |     locales: string | string[],
  38 |     options: KnownOptions
  39 |   ) => Intl.DateTimeFormat | Intl.RelativeTimeFormat | Intl.NumberFormat | IntlMessageFormat
  
  
[error.dump.d9b7032cfbd9d96a45897e8f84b3c9bc.log](https://github.com/ember-intl/ember-intl/files/5918736/error.dump.d9b7032cfbd9d96a45897e8f84b3c9bc.log)

**Comment by mabab** [comment]
dont work with last  @babel/core : 7.12.13

fix by yarn: 

"resolutions": {
    "@babel/core": "7.12.10"
},

**Comment by sinankeskin** [comment]
Same here.

**Comment by longlho** [comment]
fixed by #1505 as well

**Comment by Techn1x** [comment]
@longlho  I'm not sure that it is fixed

The yarn resolution above works (but not a permanent fix)

When I bump ember-intil to latest commit `git+https://github.com/ember-intl/ember-intl.git#917c1f1` I get this (very similar) error

```
Build Error (broccoli-persistent-filter:Babel > [Babel: ember-intl]) in ember-intl/-private/formatters/-base.ts

/home/abc/my-app/ember-intl/-private/formatters/-base.ts: Abstract methods can only appear within an abstract class. (39:2)

  37 |   }
  38 |
> 39 |   abstract format<T>(
     |   ^
  40 |     locale: string | string[],
  41 |     value: T,
  42 |     formatOptions?: KnownOptions & BaseOptions
```

**Comment by sinankeskin** [comment]
Hi @Techn1x

I don't use yarn so yarn is not my case and also I didn't use this PR but this issue resolved in my app.

I just upgraded all packages in my app with [npm-check-updates](https://github.com/raineorshine/npm-check-updates) and now it builds flawlessly.

Typescript is my dev dependency and I guess this issue resolved because of the typescript version upgrade. So in my opinion, it has nothing with the PR and babel.

My typescript version is now: v4.1.5

**Comment by Techn1x** [comment]
Fixed in `@babel/core` v7.10.14
https://github.com/babel/babel/blob/main/CHANGELOG.md#v71214-2021-02-03
> fix(ts): allow abstract methods with export default abstract class

Can confirm upgrading to that solves the issue.
